### PR TITLE
Support pwa in boilerplate template

### PR DESCRIPTION
**Pull request by 0xaptosj**
Tested on both iOS and Android

### CODE REVIEW

File: templates/nextjs-boilerplate-template/package.json [hunk 1]
```diff
@@ -52,6 +53,7 @@
     "typescript": "^5.2.2",
     "tree-kill": "1.2.2",
     "dotenv": "16.3.1",
-    "vercel": "^35.2.4"
+    "vercel": "^35.2.4",
+    "webpack": "^5.95.0"
```
Comment (0xmaayan) [inline templates/nextjs-boilerplate-template/package.json:57]: och... is that a must?

Comment (0xaptosj) [reply templates/nextjs-boilerplate-template/package.json:57]: i think so https://ducanh-next-pwa.vercel.app/docs/next-pwa/getting-started

**Comment by 0xaptosj** [comment]
> overall looks good.
> 
> Why dont we add this as an option to the wizard? same as we do for Surf? We auto install bunch of things that are not needed for a non-PWA

@0xmaayan i think PWA is always good to have

File: templates/nextjs-boilerplate-template/package.json (hunk 1 above)
Comment (0xaptosj) [reply templates/nextjs-boilerplate-template/package.json]: actuallly i removed it and it still works, remove now

File: templates/boilerplate-template/README.md [hunk 2]
```diff
@@ -22,6 +22,7 @@ To get started with the Boilerplate template and learn more about the template f
 - Aptos TS SDK
 - Aptos Wallet Adapter
 - Node based Move commands
+- [Vite-pwa](https://vite-pwa-org.netlify.app/)
```
Comment (0xmaayan) [inline templates/boilerplate-template/README.md:25]: could we add a section explaining the end user 
1. what is PWA briefly
2. once they have the boilerplate generated locally, what should they do?
3. how to deploy it?

Comment (0xaptosj) [reply templates/boilerplate-template/README.md:25]: updated doc on learn

Comment (0xaptosj) [reply templates/boilerplate-template/README.md:25]: updated doc on learn

Comment (0xaptosj) [reply templates/boilerplate-template/README.md:25]: updated doc on learn

Comment (0xaptosj) [reply templates/boilerplate-template/README.md:25]: updated doc on learn

File: templates/nextjs-boilerplate-template/next.config.mjs [hunk 3]
```diff
@@ -1,8 +1,13 @@
-/** @type {import('next').NextConfig} */
-const nextConfig = {
+import withPWAInit from "@ducanh2912/next-pwa";
+
+const withPWA = withPWAInit({
```
Comment (0xmaayan) [inline templates/nextjs-boilerplate-template/next.config.mjs:3]: is this being used?

**Review by 0xmaayan — APPROVED** [review]

File: templates/nextjs-boilerplate-template/next.config.mjs [hunk 4]
```diff
@@ -1,8 +1,13 @@
...
+  dest: "public",
```
Comment (0xaptosj) [inline templates/nextjs-boilerplate-template/next.config.mjs:4]:
```suggestion
```

**Pull request merged**